components/builder-depot/src/server.rs

```diff
@@ -326,7 +326,12 @@ fn remove_file(filename: &PathBuf) -> Result<bool> {
 }
 
 fn upload_origin_key(req: &mut Request) -> IronResult<Response> {
-    let depot = req.get::<persistent::Read<Depot>>().unwrap();
+    // JB TODO: the following two lines are a temporary hack for the channels work, because we're
+    // using an in-memory HashMap to store channel data instead of a proper datastore. These should
+    // be removed (along with all of the other lines like this in this file) and persistent::Read
+    // should be put back once the depot is using PG for storage.
+    let lock = req.get::<persistent::State<Depot>>().unwrap();
```

Comment:
If I'm reading the Iron docs correctly, `req.get()` returns a `Result` where the success value is some plugin and the error value is what you get if Iron can't create/find the plugin you want.  In this case, I'm guessing we were using `unwrap()` because we're trying to retrieve the `Depot`, which we know exists because we linked it to Iron at the bottom of the `router` function.  I'm not entirely clear if plugins are always guaranteed to be returned, provided you linked them properly.  If so, then I think we're safe here.  If not, then we likely need to catch the error explicitly.

@reset Any thoughts on this?